cmd/csaf_checker/roliecheck.go

```diff
@@ -52,35 +73,87 @@ func tlpLabel(label *csaf.TLPLabel) csaf.TLPLabel {
 	return csaf.TLPLabelUnlabeled
 }
 
+// extractTLP extracts the tlp label of the given document
+// and defaults to TLP:WHITE if not found.
+func (p *processor) extractTLP(doc any) csaf.TLPLabel {
+	advisoryTLP, err := p.expr.Eval(`$.document.distribution.tlp.label`, doc)
+	if err != nil {
+		return csaf.TLPLabelUnlabeled
+	}
+	label, ok := advisoryTLP.(string)
+	if !ok {
+		return csaf.TLPLabelUnlabeled
+	}
+	return csaf.TLPLabel(label)
+}
+
+// evaluateTLP extracts the TLP label from a given document and
+// calls upon functions further checking for mistakes in access-protection
+// or label assignment between feed and advisory
+func (lc *labelChecker) evaluateTLP(p *processor, doc any, url string) {
+
+	advisoryLabel := p.extractTLP(doc)
+
+	// If the client has no authorization it shouldn't be able
+	// to access TLP:AMBER or TLP:RED advisories
+	if !p.usedAuthorizedClient() &&
+		(advisoryLabel == csaf.TLPLabelAmber || advisoryLabel == csaf.TLPLabelRed) {
+		p.badAmberRedPermissions.use()
+		p.badAmberRedPermissions.error(
+			"Advisory %s of TLP level %v is not access protected.",
+			url, advisoryLabel)
+	}
+
+	// If we found a white labeled document we need to track it
+	// to find out later if there was an unprotected way to access it.
+	if advisoryLabel == csaf.TLPLabelWhite {
+		p.invalidAdvisories.use()
```

Comment:
Good question. I've discussed this with @JanHoefelmeyer before. The `invalidAdvisory` stems from the fact that a failure in the aftermost extraction of the document identifier indicates that the document is not valid.
This could not happen as the schema validation ran before that.

I've added a comment in commit 284c0a3999f4d8003e9fc0b84925ba8b96187c36